pkg/operator/ceph/csi/spec.go

```diff
@@ -392,14 +395,27 @@ func startDrivers(clientset kubernetes.Interface, rookclientset rookclient.Inter
 		}
 	}
 
-	tp.ProvisionerReplicas = 2
+	tp.ProvisionerReplicas = defaultProvisionerReplicas
 	nodes, err := clientset.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
 	if err == nil {
 		if len(nodes.Items) == 1 {
 			tp.ProvisionerReplicas = 1
 		}
 	} else {
-		logger.Errorf("failed to get nodes. Defaulting the number of replicas of provisioner pods to 2. %v", err)
+		logger.Errorf("failed to get nodes. Defaulting the number of replicas of provisioner pods to %d. %v", tp.ProvisionerReplicas, err)
+	}
+
+	// retrieve CSI_PROVISIONER_REPLICAS only if the node count is greater than 1
+	if nodes != nil && len(nodes.Items) > 1 {
+		replicas, err := k8sutil.GetOperatorSetting(clientset, controllerutil.OperatorSettingConfigMapName, "CSI_PROVISIONER_REPLICAS", "2")
```

Comment:
How about moving lines 410-418 inside an `else` block on line 403? Then we don't need line 409 that again checks for the number of nodes. I think it will simplify the code readability.